pkg/dockerregistry/server/client/client.go

```diff
@@ -0,0 +1,152 @@
+package client
+
+import (
+	restclient "k8s.io/client-go/rest"
+	kcoreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+
+	"github.com/openshift/origin/pkg/client"
+
+	imageclientv1 "github.com/openshift/origin/pkg/image/generated/clientset/typed/image/v1"
+	userclientv1 "github.com/openshift/origin/pkg/user/generated/clientset/typed/user/v1"
+)
+
+// RegistryClient provides OpenShift and Kubernetes clients to Docker Registry.
+type RegistryClient interface {
+	// Client returns the authenticated client to use with the server.
+	Client() (Interface, error)
+	// ClientFromToken returns a client based on the user bearer token.
+	ClientFromToken(token string) (Interface, error)
+}
+
+// Interface contains client methods that registry use to communicate with
+// OpenShift or Kubernetes API.
+type Interface interface {
+	ImageSignaturesInterfacer
+	ImagesInterfacer
+	ImageStreamImagesNamespacer
+	ImageStreamMappingsNamespacer
+	ImageStreamSecretsNamespacer
+	ImageStreamsNamespacer
+	ImageStreamTagsNamespacer
+	LimitRangesGetter
+	LocalSubjectAccessReviewsNamespacer
+	SubjectAccessReviews
+	UsersInterfacer
+}
+
+type apiClient struct {
+	oc    client.Interface
+	kube  kcoreclient.CoreInterface
+	image imageclientv1.ImageV1Interface
+	user  userclientv1.UserV1Interface
+}
+
+func newAPIClient(
+	c client.Interface,
+	kc kcoreclient.CoreInterface,
+	imageClient imageclientv1.ImageV1Interface,
+	userClient userclientv1.UserV1Interface,
+) Interface {
+	return &apiClient{
+		oc:    c,
+		kube:  kc,
+		image: imageClient,
+		user:  userClient,
+	}
+}
+
+func (c *apiClient) Users() UserInterface {
+	return c.user.Users()
+}
+
+func (c *apiClient) Images() ImageInterface {
+	return c.image.Images()
+}
+
+func (c *apiClient) ImageSignatures() ImageSignatureInterface {
+	return c.image.ImageSignatures()
+}
+
+func (c *apiClient) ImageStreams(namespace string) ImageStreamInterface {
+	return c.image.ImageStreams(namespace)
+}
+
+func (c *apiClient) ImageStreamImages(namespace string) ImageStreamImageInterface {
+	return c.image.ImageStreamImages(namespace)
+}
+
+func (c *apiClient) ImageStreamMappings(namespace string) ImageStreamMappingInterface {
+	return c.image.ImageStreamMappings(namespace)
+}
+
+func (c *apiClient) ImageStreamTags(namespace string) ImageStreamTagInterface {
+	return c.image.ImageStreamTags(namespace)
+}
+
+func (c *apiClient) ImageStreamSecrets(namespace string) ImageStreamSecretInterface {
+	return c.oc.ImageStreamSecrets(namespace)
```

Comment:
@dmage we don't have client for ImageSecrets yet, so i need to keep legacy for that resource (for now). We need to fix the API to return correct resource instead of `[]Secret`.